rust/core-lib/src/view.rs

```diff
@@ -104,130 +104,30 @@ impl View {
-                        .push(sel_end_ix)
-                );
-            }
-            if line_num == cursor_line {
-                line_builder = line_builder.push_array(|builder|
-                    builder.push("cursor")
-                        .push(cursor_col)
-                );
-            }
-            builder = builder.push(line_builder.build());
-            line_num += 1;
-            if is_last_line || line_num == last_line {
-                break;
-            }
-        }
-        builder.build()
-    }
-
-    pub fn render_spans(&self, mut builder: ArrayBuilder, start: usize, end: usize) -> ArrayBuilder {
-        let style_spans = self.style_spans.subseq(Interval::new_closed_open(start, end));
-        for (iv, style) in style_spans.iter() {
-            builder = builder.push_array(|builder|
-                builder.push("fg")
-                    .push(iv.start())
-                    .push(iv.end())
-                    .push(style.fg)
-                    .push(style.font_style));
-        }
-        builder
-    }
-
     // Render a single line, and advance cursors to next line.
     fn render_line<W: Write>(&self, tab_ctx: &TabCtx<W>, text: &Rope,
-        builder: ArrayBuilder, cursor: &mut Cursor<RopeInfo>,
-        breaks_cursor: Option<&mut Cursor<BreaksInfo>>, line_num: usize) -> ArrayBuilder
+        cursor: &mut Cursor<RopeInfo>, breaks_cursor: Option<&mut Cursor<BreaksInfo>>,
+        line_num: usize) -> Value
     {
-        let mut line_builder = ObjectBuilder::new();
         let start_pos = cursor.pos();
-        let pos = match breaks_cursor {
-            Some(bc) => {
-                let pos = bc.next::<BreaksMetric>();
-                if let Some(pos) = pos {
-                    cursor.set(pos);
-                }
-                pos
-            }
-            None => cursor.next::<LinesMetric>()
-        };
-        let pos = match pos {
-            Some(pos) => pos,
-            None => {
-                text.len()
-            }
-        };
+        let pos = breaks_cursor.map_or(cursor.next::<LinesMetric>(), |bc| {
+            let pos = bc.next::<BreaksMetric>();
+            // if using breaks update cursor
+            if let Some(pos) = pos { cursor.set(pos) }
+            pos
+        }).unwrap_or(text.len());
+
         let l_str = text.slice_to_string(start_pos, pos);
-        line_builder = line_builder.insert("text", &l_str);
         let (cursor_line, cursor_col) = self.offset_to_line_col(text, self.sel_end);
-        if line_num == cursor_line {
-            line_builder = line_builder.insert_array("cursor", |builder|
-                builder.push(cursor_col)
-            );
-        }
-        let mut sel = Vec::new();
+        let cursors = match line_num == cursor_line {
+            true => {
+                let mut c = Vec::new();
+                c.push(cursor_col);
+                Some(c) 
+            },
+            false => None,
```

Comment:
These changes make the core send a `"cursor":null` attribute for every line that does not contain a cursor whereas before the `cursor` attribute would only be either an empty array or an array with one element.